### PR TITLE
Fix and unify `CreateOrUpdateXXX` functions for karmadactl

### LINKED ISSUE #2687: CreateOrUpdateSecret function not works as expected
**What would you like to be added**:
The `CreateOrUpdateSecret` doesn't work as expected as the `Update` operation would fail due to not refresh the latest `ResourceVersion`:
https://github.com/karmada-io/karmada/blob/5d42e82b7b92812a04aaade4b77e66556aa20d66/pkg/util/secret.go#L49-L61

For more details please refer to: https://github.com/karmada-io/karmada/pull/2596#discussion_r1006333298

Since this function is used by `karmadactl init` and `karmadactl register`, so we need to verify the two functionalities after we make changes.

**Why is this needed**:

### PR DESCRIPTION
**Pull request by lonelyCZ**
Signed-off-by: lonelyCZ <chengzhe@zju.edu.cn>

**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #2687 

**Special notes for your reviewer**:
/cc @RainbowMango 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
/assign 
I'll look at it tomorrow.

File: pkg/karmadactl/util/idempotency.go [hunk 1]
```diff
@@ -0,0 +1,160 @@
+package util
```
Comment (RainbowMango) [inline pkg/karmadactl/util/idempotency.go:1]: What's the meaning of the file name `idempotency.go`? 
Should we move these functions to `pkg/util`?

Comment (lonelyCZ) [reply pkg/karmadactl/util/idempotency.go:1]: This should be from https://github.com/kubernetes/kubernetes/blob/20936e51b80255bbb28a4eed5a814121dac6149e/cmd/kubeadm/app/util/apiclient/idempotency.go

I think it is more applied to `karmadactl` that doesn't call `update` frequently.

Comment (RainbowMango) [reply pkg/karmadactl/util/idempotency.go:1]: that's ok. no big deal.

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve

**Comment by RainbowMango** [comment]
As you know, the related function hasn't been covered by E2E, I assume you have verified it locally.
> Since this function is used by karmadactl init and karmadactl register, so we need to verify the two functionalities after we make changes.

**Comment by lonelyCZ** [comment]
> As you know, the related function hasn't been covered by E2E, I assume you have verified it locally.

Right, it works fine in my env.

**Pull request merged**